### PR TITLE
Fix remove actions for entry types in the editor

### LINKED ISSUE #6906: Cannot delete standard entry types
Ubuntu 20.04, JabRef 5.1
The following image shows the dialog to customize entry types. I want to delete the entry "Patent" because I never use it. I click on the trash icon. "Patent" disappears. I accept changes. I open the entry type dialog again.

Result:
"Patent" appears again

Expected behavior: 
"Patent" is deleted

Alternative behavior:
"Patent" is a standard field in bixtex, the trash icon shouldn't only appear for user defined entry types.

grafik

### PR DESCRIPTION
**Pull request by dawidm**
fixes #6906
* Created separate view models for standard and custom entry types, adding remove icon only for the custom ones
* Updated CHANGELOG.md

It's working solution, but please take a look whether the implementation is good. I will modify the user doc if it's accepted.

- [x] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed the wrong behavior that font size changes are not reflected in dialogs. [#6039](https://github.com/JabRef/jabref/issues/6039)
 - We fixed an issue where the sort order of the entry table was reset after a restart of JabRef. [#6898](https://github.com/JabRef/jabref/pull/6898)
 - We fixed an issue where no longer a warning was displayed when inserting references into LibreOffice with an invalid "ReferenceParagraphFormat". [#6907](https://github.com/JabRef/jabref/pull/60907).
+- We fixed an issue where remove icon was shown for custom entry types in th editor
```
Comment (Siedlerchr) [inline CHANGELOG.md:40]:
```suggestion
- We fixed an issue where a remove icon was shown for standard entry types in the custom entry types dialog [6906](https://github.com/JabRef/jabref/issues/6906)
```

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
In general looks good to me. Please fix the checkstyle tests.

File: src/main/java/org/jabref/gui/customentrytypes/StandardEntryTypeViewModel.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.jabref.gui.customentrytypes;
+
+import org.jabref.model.entry.BibEntryType;
+
+public class StandardEntryTypeViewModel extends EntryTypeViewModel {
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/customentrytypes/StandardEntryTypeViewModel.java:5]: Is this class really necessary? Wouldn't it be enough to make EntryTypeViewModel not abstract and to extend it to the CustomEntryTypeViewModel?

**Review by calixtus — COMMENTED** [review]
Great work, just a question about the class StandardEntryTypeViewModel.

**Comment by dawidm** [comment]
I've created request with user doc update: https://github.com/JabRef/user-documentation/pull/310

**Review by Siedlerchr — APPROVED** [review]
Thanks, loos good now!
For external contributions we require that at least second dev reviews the changes

File: src/main/java/org/jabref/gui/customentrytypes/StandardEntryTypeViewModel.java (hunk 2 above)
Comment (dawidm) [reply src/main/java/org/jabref/gui/customentrytypes/StandardEntryTypeViewModel.java:5]: Yeah, I also though of that. I can change it.

**Review by calixtus — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**